src/openassetio-core/include/openassetio/errors/exceptions.hpp

```diff
@@ -28,8 +28,8 @@ struct OPENASSETIO_CORE_EXPORT OpenAssetIOException : std::runtime_error {
 };
 
 /**
- * Thrown whenever a procedure must abort due to an invalid input being
- * provided as an input state.
+ * Thrown whenever the input to a public API function is invalid for the
```

Comment:
[Consider] We've used the term `procedure` in all the other docstrings, whereas now this one uses `function`. I went with `procedure` for ... I dunno, silly reasons, I think it's even less technically correct in this context. Should we just change them all over to `function` for consistency?

Reply:
I think in this case it was more to be super clear that this is meant for validation of API function arguments, rather than  problems that happen during processing

Reply:
Fair enough.